src/mainviews/create-job.tsx

```diff
@@ -58,10 +58,26 @@ export function CreateJob(props: ICreateJobProps): JSX.Element {
     {}
   );
 
+  // Cache environment list.
+  const [environmentList, setEnvironmentList] = useState<
+    Scheduler.IRuntimeEnvironment[]
+  >([]);
+
   // A mapping from input names to error messages.
   // If an error message is "truthy" (i.e., not null or ''), we should display the
   // input in an error state and block form submission.
-  const [errors, setErrors] = React.useState<SchedulerTokens.ErrorsType>({});
+  const [errors, setErrors] = useState<SchedulerTokens.ErrorsType>({});
+
+  const api = new SchedulerService({});
```

Comment:
This creates a new `SchedulerService` every render, which is especially bad for forms where a render is triggered every time a key is pressed.

Right now, the SS constructor isn't doing much, so this doesn't have too much of an impact yet. But let's not have our form performance have an implicit dependency on the SS constructor.

```
const api = useMemo(() => new SchedulerService({}), []);
```